### ISSUE TITLE
b:radioButton required

### ISSUE THREAD

**Issue by chongma** [TheCoder4eu/BootsFaces-OSP#648]
tried to use a b:radioButton instead of h:selectOneMenu in the 
```xml
<b:radiobutton value="#{radiobuttonBean.favoriteColor}">
      <f:selectItem itemValue="Red"   itemLabel="Red it is!" />
      <f:selectItem itemValue="Green" itemLabel="I love green colors." />
      <f:selectItem itemValue="Blue"  itemLabel="What about blue?" />
  </b:radiobutton>
```
 configuration but the b:radioButton lacks a required attribute or it is not working properly allowing empty submission

**Comment by stephanrauh** [comment]
It's a bit difficult to implement the `required` attribute with radio buttons because the radio buttons are only linked on the client. But maybe it's possible. Let us ponder about it.

**Comment by stephanrauh** [comment]
To my surprise, the `required` attribute already was supported, but a lot of corner cases didn't work yet. I've fixed that. BTW, I've also fixed AJAX along the way. For some reason unknown, it was broken.

![image](https://cloud.githubusercontent.com/assets/3045767/24082882/f35431d8-0ccc-11e7-8d19-260923dcad5e.png)

![image](https://cloud.githubusercontent.com/assets/3045767/24082902/608fe350-0ccd-11e7-85dc-33d95c5eef1b.png)
